src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -1592,13 +1592,14 @@ def _install_dev_spaces_extension(extension_name):
 def _update_dev_spaces_extension(extension_name, extension_module):
     from azure.cli.core.extension import ExtensionNotInstalledException
     try:
-        # unloading the imported module to update
-        del sys.modules[extension_module]
-
         from azure.cli.command_modules.extension import custom
         custom.update_extension(extension_name=extension_name)
+
+        # reloading the imported module to update
+        from importlib import reload as reload
+        reload(sys.modules[extension_module])
```

Comment:
If you want this to also work for python2 you can do something like:

```
try:
    from importlib import reload as reload
except ImportError:
    pass
reload(module)
```

Reply:
Oh, I thought it worked for Python2 as well as it passed CI. Updating ...

Reply:
`reload()` is a builtin function for python2, hence the try-catch around the import.